### ISSUE TITLE
Supports "Ergonomic brand checks for Private Fields" in TypeScript

### ISSUE THREAD

**Issue by ayame113** [denoland/deno#12565]
It seems that "[Ergonomic brand checks for Private Fields](https://github.com/tc39/proposal-private-fields-in-in)", which is stage 4 in TC39, is not yet available in Deno.

The following code in a [v8 blog post](https://v8.dev/blog/v8-release-91#private-brand-checks-a.k.a.-%23foo-in-obj) works on chrome, but not on Deno.
```ts
class A {
  static test(obj: any) {
    console.log(#foo in obj);
  }
  #foo = 0;
}

A.test(new A()); // true
A.test({}); // false
```

chrome:
![image](https://user-images.githubusercontent.com/40050810/139122580-2c025766-fbc4-4cce-b21f-d4fcb0fdb983.png)

Deno:
![image](https://user-images.githubusercontent.com/40050810/139122881-968972ad-9ea2-4cbd-a824-99df88c62ef9.png)

The version used is as follows.
```
>deno --version
deno 1.15.3+b44b26c (canary, x86_64-pc-windows-msvc)
v8 9.5.172.19
typescript 4.4.2
```

**Comment by kitsonk** [comment]
Still need #12410 to complete, reopening.

**Comment by kitsonk** [comment]
@dsherret swc/deno_ast doesn't seem to support it.  The test I added for TS 4.5 (https://github.com/denoland/deno/pull/12410/files#diff-e4aa5f7d4bbea624f0785cf9736d677d65a2f4e255ef7163e3313551f715a881) is failing to even parse, so not even getting to tsc.

**Comment by dsherret** [comment]
@kitsonk hmmm looks like 0.80.0 (swc_ecmascript version we're on) had a dependency on 0.75.0 of swc_ecma_parser and the fix was published in 0.76.1 of that, but an swc_ecmascript version bump didn't occur until a few days later. I will update deno_ast.
